src/asteroid_tracker/asteroid_tracker.py

```diff
@@ -13,50 +16,122 @@
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
-
+format_Error = False
 
 API_URL = 'https://api.nasa.gov/neo/rest/v1/feed?start_date=%s&end_date=%s&api_key=DEMO_KEY'
 
 
 def main():
     """
-    Main function which parses input and calculates statistics.
+    Main function which takes input arguments and calls relevant functions.
     """
 
     start_date = None
     end_date = None
 
     # TODO: Task 1 - Gather user input
+    start_date, end_date = input(" ").split()
+    # start_date = '2021-01-01'
+    # end_date = 'fff'
 
-    stats = calculate_statistics(start_date, end_date)
 
+    # start_date
+    # year, month, day = map(int, start_date.split('-'))
+    # start_date = datetime.date(year, month, day)
+
+    # start_date = start_date.strip()
+
+    # end_date
+    # year, month, day = map(int, end_date.split('-'))
+    # end_date = datetime.date(year, month, day)
+
+    # end_date = end_date.strip()
+
+    stats = calculate_statistics(start_date, end_date)
     print_statistics(stats)
 
 
 def calculate_statistics(start_date, end_date):
     """
     Make an API request and calculate statistics.
     """
-
     # TODO: Task 2 - Prepare and make the HTTP request
+    global format_Error
+    try:
+        datetime.datetime.strptime(start_date, '%Y-%m-%d')
+    except ValueError:
+        format_Error = True
 
+    try:
+        datetime.datetime.strptime(end_date, '%Y-%m-%d')
+    except ValueError:
+        format_Error = True
 
-    num_asteroids = 0
-    num_potentially_hazardous_asteroids = 0
+    d_error = {'code': 400, 'type': 'BAD_REQUEST', 'message': 'Date Format Exception - Expected format (yyyy-mm-dd) - The Feed date limit is only 7 Days', }
+    dict_error = {'error': d_error}
 
-    largest_diameter_meters = -1
-    nearest_miss_kms = -1
+    if format_Error:
+        return dict_error
 
-    # TODO: Task 3 - Calculate statistics
-
-    return {
-        'start_date': start_date,
-        'end_date': end_date,
-        'num_asteroids': num_asteroids,
-        'num_potentially_hazardous_asteroids': num_potentially_hazardous_asteroids,
-        'largest_diameter_meters': largest_diameter_meters,
-        'nearest_miss_kms': nearest_miss_kms,
-    }
+    else:
+        url = API_URL % (start_date, end_date)
+        response = requests.get(url)
+        # print(response.content)
+
+        json_data = json.loads(response.content)
+        # print(json.dumps(json_data, indent = 4))
+
+        # num_asteroids
+        keyVal = 'element_count'
+        if keyVal in json_data:
+            num_asteroids = json_data[keyVal]
+
+        # num_potentially_hazardous_asteroids
+        num_potentially_hazardous_asteroids = 0
+        count = 0
+
+        for nearEarthObject in json_data['near_earth_objects']:
```

Comment:
There is a mix of snake_case, camelCase and whatever case format_Error is throughout the code. It's important to be consistent with this sort of thing to improve readability of the code, and specifically in Python where snake_case is used for almost everything except class names.